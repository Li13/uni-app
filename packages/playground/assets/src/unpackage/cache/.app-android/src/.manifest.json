{
  "version": "1",
  "env": {
    "compiler_version": "4.44.2024122002-dev"
  },
  "files": {
    "pages/index/index.kt": {
      "md5": "02c543da699e39e48dabce26f5433e98838af8e6",
      "class": "GenPagesIndexIndex"
    },
    "index.kt": {
      "md5": "841ad2bff78f405111a50e590ae55e7633715dfc",
      "class": ""
    },
    "uni_modules/test-com/components/test-com/test-com.kt": {
      "md5": "edbe211a9840230e157636f5f71a22ff9e6e48e8",
      "class": "GenUniModulesTestComComponentsTestComTestCom"
    },
    "pages/sub/test/test.kt": {
      "class": "GenPagesSubTestTest",
      "md5": "669a359b6cd7b1ea7c3f088a2ffcc2f2fa0c920b"
    }
  }
}