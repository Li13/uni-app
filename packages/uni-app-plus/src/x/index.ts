import * as uni from './api/index'
import { registerApp as __registerApp, initApp } from './framework/app'
import {
  definePage as __definePage,
  getCurrentPages,
} from '../service/framework/page'
import * as components from './components'

import { registerSystemRoute } from './framework/route'
import { systemRoutes as __uniSystemRoutes } from './framework/route'

import UniChooseLocationPage from '@dcloudio/uni-ext-api/uni-chooseLocation/pages/chooseLocation/chooseLocation.vue'
registerSystemRoute('uni:chooseLocation', UniChooseLocationPage, {
  disableSwipeBack: false,
})

import UniActionSheetPage from '@dcloudio/uni-ext-api/uni-actionSheet/pages/actionSheet/actionSheet.vue'
registerSystemRoute('uni:actionSheet', UniActionSheetPage, {
  disableSwipeBack: false,
})
import {
  defineAsyncApi,
  defineOffApi,
  defineOnApi,
  defineSyncApi,
  defineTaskApi,
} from '@dcloudio/uni-api'

export {
  uni,
  getCurrentPages,
  __definePage,
  __registerApp,
  __uniSystemRoutes,
  initApp,
  components,
  defineAsyncApi,
  defineOffApi,
  defineOnApi,
  defineSyncApi,
  defineTaskApi,
}
