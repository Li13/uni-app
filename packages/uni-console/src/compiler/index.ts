import type { Plugin } from 'vite'
import path from 'path'
import {
  defineUniMainJsPlugin,
  isEnableConsole,
  resolveBuiltIn,
} from '@dcloudio/uni-cli-shared'

const uniConsoleRuntimePlugin = (): Plugin => {
  return {
    name: 'uni:console:runtime',
    config() {
      const isProd = process.env.NODE_ENV === 'production'
      return {
        define: {
          __UNI_SOCKET_HOSTS__: JSON.stringify(
            isProd ? '' : process.env.UNI_SOCKET_HOSTS
          ),
          __UNI_SOCKET_PORT__: JSON.stringify(
            isProd ? '' : process.env.UNI_SOCKET_PORT
          ),
          __UNI_SOCKET_ID__: JSON.stringify(
            isProd ? '' : process.env.UNI_SOCKET_ID
          ),
        },
      }
    },
  }
}

export default () => {
  return [
    uniConsoleRuntimePlugin(),
    defineUniMainJsPlugin((opts) => {
      const hasRuntimeSocket = isEnableConsole()
      return {
        name: 'uni:console-main-js',
        enforce: 'post',
        resolveId(id: string) {
          if (id === '@dcloudio/uni-console') {
            return resolveBuiltIn(
              path.join('@dcloudio/uni-console', 'dist/index.esm.js')
            )
          }
        },
        transform(code: string, id: string) {
          if (!hasRuntimeSocket) {
            return
          }
          if (!opts.filter(id)) {
            return
          }
          return {
            code: `import '@dcloudio/uni-console'
            ${code}
            `,
            map: {
              mappings: '',
            },
          }
        },
      }
    }),
  ]
}
