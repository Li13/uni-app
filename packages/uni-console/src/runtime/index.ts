import { hasRuntimeSocket, initRuntimeSocket } from './socket'
import { rewriteConsole, setSend } from './console'

export function initConsole(): Promise<boolean> {
  if (!hasRuntimeSocket) return Promise.resolve(false)

  const restoreConsole = rewriteConsole()
  return initRuntimeSocket().then((socket) => {
    if (!socket) {
      restoreConsole()
      console.error('开发模式下日志通道建立连接失败')
      return false
    }
    setSend((msgs: any) => {
      socket!.send({
        data: {
          type: 'console',
          data: msgs,
        },
      })
    })
    return true
  })
}

initConsole()
