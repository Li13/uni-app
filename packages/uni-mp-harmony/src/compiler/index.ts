import type { Plugin } from 'vite'
import { ASSETS_INLINE_LIMIT } from '@dcloudio/uni-cli-shared'
import initMiniProgramPlugin from '@dcloudio/uni-mp-vite'
import { options } from './options'

const uniMiniProgramHarmonyPlugin: Plugin = {
  name: 'uni:mp-harmony',
  config() {
    return {
      define: {
        __VUE_CREATED_DEFERRED__: true,
      },
      build: {
        // css 中不支持引用本地资源
        assetsInlineLimit: ASSETS_INLINE_LIMIT,
      },
    }
  },
}

export default [uniMiniProgramHarmonyPlugin, ...initMiniProgramPlugin(options)]
