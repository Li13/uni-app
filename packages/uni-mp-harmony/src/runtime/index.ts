import { EventChannel } from '@dcloudio/uni-shared'
import {
  createApp,
  createComponent,
  createPage,
  createSubpackageApp,
} from '@dcloudio/uni-quickapp-webview/src/runtime'

export * from '@dcloudio/uni-quickapp-webview/src/runtime'
;(has as any).EventChannel = EventChannel
;(has as any).createApp = (global as any).createApp = createApp
;(has as any).createPage = createPage
;(has as any).createComponent = createComponent
;(has as any).createSubpackageApp = (global as any).createSubpackageApp =
  createSubpackageApp
