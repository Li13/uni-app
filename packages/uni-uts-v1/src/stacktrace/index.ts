import { relative } from '../utils'
import {
  type GenerateAppIOSJavaScriptRuntimeCodeFrameOptions,
  type GenerateJavaScriptRuntimeCodeFrameOptions,
  parseUTSJavaScriptRuntimeStacktrace,
} from './js'
import {
  type GenerateAppAndroidKotlinRuntimeCodeFrameOptions,
  type GenerateKotlinRuntimeCodeFrameOptions,
  parseUTSKotlinRuntimeStacktrace,
} from './kotlin'
import {
  type GenerateWeixinRuntimeCodeFrameOptions,
  parseWeixinRuntimeStacktrace,
} from './mp/weixin'

export { parseUTSSwiftPluginStacktrace } from './swift'

export {
  parseUTSKotlinStacktrace,
  parseUTSKotlinRuntimeStacktrace,
} from './kotlin'

export { parseUTSJavaScriptRuntimeStacktrace } from './js'

export function parseRuntimeStacktrace(
  stacktrace: string,
  options:
    | GenerateAppAndroidKotlinRuntimeCodeFrameOptions
    | GenerateAppIOSJavaScriptRuntimeCodeFrameOptions
    | GenerateWeixinRuntimeCodeFrameOptions
) {
  if (options.platform === 'mp-weixin') {
    return parseWeixinRuntimeStacktrace(stacktrace, options)
  }
  return parseUTSRuntimeStacktrace(stacktrace, options)
}

export function parseUTSRuntimeStacktrace(
  stacktrace: string,
  options:
    | GenerateKotlinRuntimeCodeFrameOptions
    | GenerateJavaScriptRuntimeCodeFrameOptions
) {
  if (options.language === 'kotlin') {
    return parseUTSKotlinRuntimeStacktrace(stacktrace, options)
  } else if (options.language === 'javascript') {
    return parseUTSJavaScriptRuntimeStacktrace(stacktrace, options)
  }
}

export function parseUTSSyntaxError(error: any, inputDir: string): string {
  if (error instanceof Error) {
    error = error.message
  }
  let msg = String(error).replace(/\t/g, ' ')
  let res: RegExpExecArray | null = null
  const syntaxErrorRe = /(,-\[(.*):(\d+):(\d+)\])/g
  while ((res = syntaxErrorRe.exec(msg))) {
    const [row, filename, line, column] = res.slice(1)
    msg = msg.replace(
      row,
      `at ${relative(filename.split('?')[0], inputDir)}:${
        parseInt(line) + 3
      }:${column}`
    )
  }
  return msg
}
