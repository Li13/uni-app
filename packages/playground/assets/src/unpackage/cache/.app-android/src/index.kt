@file:Suppress("UNCHECKED_CAST", "USELESS_CAST", "INAPPLICABLE_JVM_NAME", "UNUSED_ANONYMOUS_PARAMETER")
package uni.UNIXXXXXXX
import io.dcloud.uniapp.*
import io.dcloud.uniapp.extapi.*
import io.dcloud.uniapp.framework.*
import io.dcloud.uniapp.runtime.*
import io.dcloud.uniapp.vue.*
import io.dcloud.uniapp.vue.shared.*
import io.dcloud.unicloud.*
import io.dcloud.uts.*
import io.dcloud.uts.Map
import io.dcloud.uts.Set
import io.dcloud.uts.UTSAndroid
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Deferred
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.async
val runBlock1 = run {
    __uniConfig.getAppStyles = fun(): Map<String, Map<String, Map<String, Any>>> {
        return GenApp.styles
    }
}
open class GenApp : BaseApp {
    constructor(__ins: ComponentInternalInstance) : super(__ins) {
        onLaunch(fun(_: OnLaunchOptions) {}, __ins)
    }
    companion object {
        val styles: Map<String, Map<String, Map<String, Any>>> by lazy {
            normalizeCssStyles(utsArrayOf())
        }
    }
}
val GenAppClass = CreateVueAppComponent(GenApp::class.java, fun(): VueComponentOptions {
    return VueComponentOptions(type = "app", name = "", inheritAttrs = true, inject = Map(), props = Map(), propsNeedCastKeys = utsArrayOf(), emits = Map(), components = Map(), styles = GenApp.styles)
}
, fun(instance): GenApp {
    return GenApp(instance)
}
)
val `default` = "/assets/pages-index-static_invalid-static.34ae5d3b.jpg"
val default1 = "/static/static_logo.png"
val default2 = "/uni_modules/test-com/static/uni_modules-test-com-static_shuijiao.jpg"
val default3 = "/static/static_logo.png"
val GenUniModulesTestComComponentsTestComTestComClass = CreateVueComponent(GenUniModulesTestComComponentsTestComTestCom::class.java, fun(): VueComponentOptions {
    return VueComponentOptions(type = "component", name = "", inheritAttrs = GenUniModulesTestComComponentsTestComTestCom.inheritAttrs, inject = GenUniModulesTestComComponentsTestComTestCom.inject, props = GenUniModulesTestComComponentsTestComTestCom.props, propsNeedCastKeys = GenUniModulesTestComComponentsTestComTestCom.propsNeedCastKeys, emits = GenUniModulesTestComComponentsTestComTestCom.emits, components = GenUniModulesTestComComponentsTestComTestCom.components, styles = GenUniModulesTestComComponentsTestComTestCom.styles)
}
, fun(instance): GenUniModulesTestComComponentsTestComTestCom {
    return GenUniModulesTestComComponentsTestComTestCom(instance)
}
)
val GenPagesIndexIndexClass = CreateVueComponent(GenPagesIndexIndex::class.java, fun(): VueComponentOptions {
    return VueComponentOptions(type = "page", name = "", inheritAttrs = GenPagesIndexIndex.inheritAttrs, inject = GenPagesIndexIndex.inject, props = GenPagesIndexIndex.props, propsNeedCastKeys = GenPagesIndexIndex.propsNeedCastKeys, emits = GenPagesIndexIndex.emits, components = GenPagesIndexIndex.components, styles = GenPagesIndexIndex.styles)
}
, fun(instance): GenPagesIndexIndex {
    return GenPagesIndexIndex(instance)
}
)
val default4 = "/pages/sub/static/pages-sub-static_2.jpg"
val GenPagesSubTestTestClass = CreateVueComponent(GenPagesSubTestTest::class.java, fun(): VueComponentOptions {
    return VueComponentOptions(type = "page", name = "", inheritAttrs = GenPagesSubTestTest.inheritAttrs, inject = GenPagesSubTestTest.inject, props = GenPagesSubTestTest.props, propsNeedCastKeys = GenPagesSubTestTest.propsNeedCastKeys, emits = GenPagesSubTestTest.emits, components = GenPagesSubTestTest.components, styles = GenPagesSubTestTest.styles)
}
, fun(instance): GenPagesSubTestTest {
    return GenPagesSubTestTest(instance)
}
)
fun createApp(): UTSJSONObject {
    val app = createSSRApp(GenAppClass)
    return UTSJSONObject(Map<String, Any?>(utsArrayOf(
        utsArrayOf(
            "app",
            app
        )
    )))
}
fun main(app: IApp) {
    definePageRoutes()
    defineAppConfig()
    (createApp()["app"] as VueApp).mount(app, GenUniApp())
}
open class UniAppConfig : io.dcloud.uniapp.appframe.AppConfig {
    override var name: String = "assets"
    override var appid: String = "__UNI__XXXXXXX"
    override var versionName: String = "1.0.0"
    override var versionCode: String = "100"
    override var uniCompilerVersion: String = "4.44"
    constructor() : super() {}
}
fun definePageRoutes() {
    __uniRoutes.push(UniPageRoute(path = "pages/index/index", component = GenPagesIndexIndexClass, meta = UniPageMeta(isQuit = true), style = utsMapOf()))
    __uniRoutes.push(UniPageRoute(path = "pages/sub/test/test", component = GenPagesSubTestTestClass, meta = UniPageMeta(isQuit = false), style = utsMapOf()))
}
val __uniLaunchPage: Map<String, Any?> = utsMapOf("url" to "pages/index/index", "style" to utsMapOf<String, Any?>())
fun defineAppConfig() {
    __uniConfig.entryPagePath = "/pages/index/index"
    __uniConfig.globalStyle = utsMapOf()
    __uniConfig.getTabBarConfig = fun(): Map<String, Any>? {
        return null
    }
    __uniConfig.tabBar = __uniConfig.getTabBarConfig()
    __uniConfig.conditionUrl = ""
    __uniConfig.uniIdRouter = Map()
    __uniConfig.ready = true
}
open class GenUniApp : UniAppImpl() {
    open val vm: GenApp?
        get() {
            return getAppVm() as GenApp?
        }
    open val `$vm`: GenApp?
        get() {
            return getAppVm() as GenApp?
        }
}
fun getApp(): GenUniApp {
    return getUniApp() as GenUniApp
}
