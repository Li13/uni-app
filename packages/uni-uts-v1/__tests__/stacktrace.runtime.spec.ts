import path from 'path'

import { parseRuntimeStacktrace } from '../src/stacktrace'

describe('uts:stacktrace:runtime', () => {
  test('parseUTSKotlinRuntimeStacktrace', async () => {
    const cacheDir = path.resolve(
      __dirname,
      'examples/uni-app-x/output/cache/app-android'
    )
    expect(
      parseRuntimeStacktrace(
        `java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
at java.util.ArrayList.get(ArrayList.java:437)
at uni.UNIXXXXXXX.GenPagesIndexIndex$1.invoke(index.kt:24)
at java.lang.reflect.Method.invoke(Native Method)
at io.dcloud.uniapp.vue.shared.IndexKt$callFunction$invoke$1.invoke(index.kt:627)
at io.dcloud.uniapp.vue.shared.IndexKt.callFunction(index.kt:638)`,
        {
          platform: 'app-android',
          language: 'kotlin',
          appid: '__UNI__XXXXXXX',
          cacheDir,
        }
      )
    ).toMatchSnapshot()
    expect(
      parseRuntimeStacktrace(
        `java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
at java.util.ArrayList.get(ArrayList.java:437)
at uni.UNIXXXXXXX.GenPagesIndexIndex$click$1.invoke(index.kt:59)
at java.lang.reflect.Method.invoke(Native Method)
at io.dcloud.uniapp.vue.shared.IndexKt$callFunction$invoke$1.invoke(index.kt:627)
at io.dcloud.uniapp.vue.shared.IndexKt.callFunction(index.kt:638)`,
        {
          platform: 'app-android',
          language: 'kotlin',
          appid: '__UNI__XXXXXXX',
          cacheDir,
        }
      )
    ).toMatchSnapshot()
    expect(
      parseRuntimeStacktrace(
        `java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
at java.util.ArrayList.get(ArrayList.java:437)
at uni.UNIXXXXXXX.IndexKt.test(index.kt:40)
at uni.UNIXXXXXXX.GenPagesIndexIndex$click1$1.invoke(index.kt:63)
at java.lang.reflect.Method.invoke(Native Method)
at io.dcloud.uniapp.vue.shared.IndexKt$callFunction$invoke$1.invoke(index.kt:627)
at io.dcloud.uniapp.vue.shared.IndexKt.callFunction(index.kt:638)`,
        {
          platform: 'app-android',
          language: 'kotlin',
          appid: '__UNI__XXXXXXX',
          cacheDir,
        }
      )
    ).toMatchSnapshot()
    expect(
      parseRuntimeStacktrace(
        `java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
at java.util.ArrayList.get(ArrayList.java:437)
at uni.UNIXXXXXXX.IndexKt.test(index.kt:40)`,
        {
          platform: 'app-android',
          language: 'kotlin',
          appid: '__UNI__XXXXXXX',
          cacheDir,
          logType: 'warn',
        }
      )
    ).toMatchSnapshot()
    expect(
      parseRuntimeStacktrace(
        `java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
at java.util.ArrayList.get(ArrayList.java:437)
at uni.UNIXXXXXXX.IndexKt.test(index.kt:40)`,
        {
          platform: 'app-android',
          language: 'kotlin',
          appid: '__UNI__XXXXXXX',
          cacheDir,
          logType: 'error',
        }
      )
    ).toMatchSnapshot()

    expect(
      parseRuntimeStacktrace(
        `java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Boolean
at uni.UNIXXXXXXX.GenComponentsTestBooleanTestBoolean.getDisabled1(test-boolean.kt:68)
at uni.UNIXXXXXXX.GenComponentsTestBooleanTestBoolean$1.invoke(test-boolean.kt:21)
at java.lang.reflect.Method.invoke(Native Method)
at io.dcloud.uniapp.vue.shared.IndexKt$callFunction$invoke$1.invoke(index.kt:708)`,
        {
          platform: 'app-android',
          language: 'kotlin',
          appid: '__UNI__XXXXXXX',
          cacheDir,
          logType: 'error',
        }
      )
    ).toMatchSnapshot()
  })
  test('parseUTSJavaScriptRuntimeStacktrace', async () => {
    const cacheDir = path.resolve(
      __dirname,
      'examples/uni-app-x/output/cache/app-ios'
    )
    expect(
      parseRuntimeStacktrace(
        `app-service.js(5:60) ReferenceError:Can't find variable: a @app-service.js:5:60`,
        {
          platform: 'app-ios',
          language: 'javascript',
          cacheDir,
        }
      )
    ).toMatchSnapshot()
    expect(
      parseRuntimeStacktrace(
        `at <Index __pageId=1 __pagePath="pages/index/index" __pageQuery=  ... >
Can't find variable: a
onLoad@app-service.js:9:64
callWithErrorHandling@uni-app-x-framework.js:2279:23
callWithAsyncErrorHandling@uni-app-x-framework.js:2286:38
@uni-app-x-framework.js:4763:45`,
        {
          platform: 'app-ios',
          language: 'javascript',
          cacheDir,
        }
      )
    ).toMatchSnapshot()
  })

  test('parseWeixinRuntimeStacktrace', async () => {
    const outputDir = path.resolve(
      __dirname,
      'examples/uni-app-x/output/dist/dev/mp-weixin'
    )
    expect(
      parseRuntimeStacktrace(
        `Error: click2
at Proxy.click2 (http://127.0.0.1:22515/appservice/pages/index/index.js:23:13)
at http://127.0.0.1:22515/appservice/pages/index/index.js:36:49
at callWithErrorHandling (http://127.0.0.1:22515/appservice/common/vendor.js:2135:22)
at callWithAsyncErrorHandling (http://127.0.0.1:22515/appservice/common/vendor.js:2142:15)
at Function.invoke (http://127.0.0.1:22515/appservice/common/vendor.js:5498:14)
at f (http://127.0.0.1:22515/appservice/__dev__/WASubContext.js?t=wechat&s=1732624710099&v=3.6.6:1:147076)
at http://127.0.0.1:22515/appservice/__dev__/WASubContext.js?t=wechat&s=1732624710099&v=3.6.6:1:147519`,
        {
          outputDir,
          platform: 'mp-weixin',
          language: 'javascript',
          cacheDir: '',
        }
      )
    ).toMatchSnapshot()
    expect(
      parseRuntimeStacktrace(
        `MiniProgramError
bbbbbbbbbbb12
Error: bbbbbbbbbbb12
at http://127.0.0.1:37922/appservice/pages/index/index.js:23:13
at http://127.0.0.1:37922/appservice/__dev__/WASubContext.js?t=wechat&s=1732621970683&v=3.6.1:1:512000
at p.runWith (http://127.0.0.1:37922/appservice/__dev__/WASubContext.js?t=wechat&s=1732621970683&v=3.6.1:1:501622)
at q (http://127.0.0.1:37922/appservice/__dev__/WASubContext.js?t=wechat&s=1732621970683&v=3.6.1:1:511978)
at ide:///package/__APP__/appservice.js:1111:7
at t.doWhenAllScriptLoaded (ide:///getmainpackage.js:1174:21)`,
        {
          outputDir,
          platform: 'mp-weixin',
          language: 'javascript',
          cacheDir: '',
        }
      )
    ).toMatchSnapshot()
  })
})
