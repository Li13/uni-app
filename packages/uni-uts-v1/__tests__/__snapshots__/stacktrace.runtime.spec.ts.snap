// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uts:stacktrace:runtime parseBaiduRuntimeStacktrace 1`] = `
"Error: click2
at Proxy.click2 (http://127.0.0.1:22515/output/pages/index/index.js:21:13)
at http://127.0.0.1:22515/output/pages/index/index.js:36:49"
`;

exports[`uts:stacktrace:runtime parseUTSJavaScriptRuntimeStacktrace 1`] = `
"error: [EXCEPTION] ReferenceError:Can't find variable: a
at pages/index/index.uvue:3:8
1  |  <script lang="uts">
2  |      setTimeout(() => {
3  |          uni.__log__('log','at pages/index/index.uvue:3',a)
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
4  |      }, 1000)
5  |      export default {"
`;

exports[`uts:stacktrace:runtime parseUTSJavaScriptRuntimeStacktrace 2`] = `
"error: [EXCEPTION] at <Index __pageId=1 __pagePath="pages/index/index" __pageQuery=  ... >
Can't find variable: a
at pages/index/index.uvue:7:12
5  |      export default {
6  |          onLoad() {
7  |              uni.__log__('log','at pages/index/index.uvue:7',a)
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
8  |          }
9  |      }"
`;

exports[`uts:stacktrace:runtime parseUTSKotlinRuntimeStacktrace 1`] = `
"error: java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
at pages/index/index.uvue:13:12
11 |          onLoad() {
12 |              const arr = [1]
13 |              console.log(arr[1])
   |              ^^^^^^^^^^^^^^^^^^^
14 |          },
15 |          methods: {"
`;

exports[`uts:stacktrace:runtime parseUTSKotlinRuntimeStacktrace 2`] = `
"error: java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
at pages/index/index.uvue:18:16
16 |              click() {
17 |                  const arr = [1]
18 |                  console.log(arr[1])
   |                  ^^^^^^^^^^^^^^^^^^^
19 |              },
20 |              click1() {"
`;

exports[`uts:stacktrace:runtime parseUTSKotlinRuntimeStacktrace 3`] = `
"error: java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
at pages/index/test.uts:3:4
1  |  export function test() {
2  |      const arr = [1]
3  |      console.log(arr[1])
   |      ^^^^^^^^^^^^^^^^^^^
4  |  }"
`;

exports[`uts:stacktrace:runtime parseUTSKotlinRuntimeStacktrace 4`] = `
"​error: java.lang.IndexOutOfBoundsException: Index: 1, Size: 1​
at pages/index/test.uts:3:4
1  |  export function test() {
2  |      const arr = [1]
3  |      console.log(arr[1])
   |      ^^^^^^^^^^^^^^^^^^^
4  |  }"
`;

exports[`uts:stacktrace:runtime parseUTSKotlinRuntimeStacktrace 5`] = `
"‌error: java.lang.IndexOutOfBoundsException: Index: 1, Size: 1‌
at pages/index/test.uts:3:4
1  |  export function test() {
2  |      const arr = [1]
3  |      console.log(arr[1])
   |      ^^^^^^^^^^^^^^^^^^^
4  |  }"
`;

exports[`uts:stacktrace:runtime parseUTSKotlinRuntimeStacktrace 6`] = `
"‌error: java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Boolean‌
at components/test-boolean/test-boolean.vue:19:12
17 |          },
18 |          created(){
19 |              console.log(this.disabled1)
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
20 |          }
21 |      }"
`;

exports[`uts:stacktrace:runtime parseWeixinRuntimeStacktrace 1`] = `
"Error: click2
at Proxy.click2 (http://127.0.0.1:22515/appservice/pages/index/index.js:23:13)
at http://127.0.0.1:22515/appservice/pages/index/index.js:36:49
at callWithErrorHandling (http://127.0.0.1:22515/appservice/common/vendor.js:2135:22)
at callWithAsyncErrorHandling (http://127.0.0.1:22515/appservice/common/vendor.js:2142:15)
at Function.invoke (http://127.0.0.1:22515/appservice/common/vendor.js:5498:14)
at f (http://127.0.0.1:22515/appservice/__dev__/WASubContext.js?t=wechat&s=1732624710099&v=3.6.6:1:147076)
at http://127.0.0.1:22515/appservice/__dev__/WASubContext.js?t=wechat&s=1732624710099&v=3.6.6:1:147519"
`;

exports[`uts:stacktrace:runtime parseWeixinRuntimeStacktrace 2`] = `
"MiniProgramError
bbbbbbbbbbb12
Error: bbbbbbbbbbb12
at http://127.0.0.1:37922/appservice/pages/index/index.js:23:13
at http://127.0.0.1:37922/appservice/__dev__/WASubContext.js?t=wechat&s=1732621970683&v=3.6.1:1:512000
at p.runWith (http://127.0.0.1:37922/appservice/__dev__/WASubContext.js?t=wechat&s=1732621970683&v=3.6.1:1:501622)
at q (http://127.0.0.1:37922/appservice/__dev__/WASubContext.js?t=wechat&s=1732621970683&v=3.6.1:1:511978)
at ide:///package/__APP__/appservice.js:1111:7
at t.doWhenAllScriptLoaded (ide:///getmainpackage.js:1174:21)"
`;
